tests/loggers/test_all.py

```diff
@@ -322,54 +330,43 @@ def test_logger_with_prefix_all(tmpdir, monkeypatch):
     """
     Test that prefix is added at the beginning of the metric keys.
     """
-    _test_loggers_with_prefix(tmpdir, TensorBoardLogger)
+    prefix = 'tmp'
 
+    # Comet
     with mock.patch('pytorch_lightning.loggers.comet.comet_ml'), \
          mock.patch('pytorch_lightning.loggers.comet.CometOfflineExperiment'):
         _patch_comet_atexit(monkeypatch)
-        _test_loggers_with_prefix(tmpdir, CometLogger)
+        logger = _instantiate_logger(CometLogger, save_idr=tmpdir, prefix=prefix)
+        logger.log_metrics({"test": 1.0}, step=0)
+        logger.experiment.log_metrics.assert_called_once_with({"tmp-test": 1.0}, epoch=None, step=0)
 
+    # MLflow
     with mock.patch('pytorch_lightning.loggers.mlflow.mlflow'), \
          mock.patch('pytorch_lightning.loggers.mlflow.MlflowClient'):
-        _test_loggers_with_prefix(tmpdir, MLFlowLogger)
+        logger = _instantiate_logger(MLFlowLogger, save_idr=tmpdir, prefix=prefix)
+        logger.log_metrics({"test": 1.0}, step=0)
+        logger.experiment.log_metric.has_calls([call(ANY, "tmp-test", 1.0, ANY, step=0)])
```

Comment:
great. Thanks, @awaelchli :smiley:. Need to study this mock tests stuff.

BTW why is this different from assertions for other loggers with `assert_called_once_with`?
Also, should it be `assert_has_calls`?

Reply:
yes, you are right. fixed it.
the `assert_called_once_with` accepts just a single call check, while `assert_has_calls` takes a list, which is useful checking the exact order of calls.